Re-structure running the workflow docs (#328)
AlexsLemonade/scpca-downstream-analyses: README.md

    - [How to install Snakemake](#how-to-install-snakemake)
  - [c) Additional dependencies](#c-additional-dependencies)
    - [Snakemake/conda installation](#snakemakeconda-installation)
- [2. Verify Input Data Format](#2-verify-input-data-format)
- [3. Create Metadata File](#3-create-metadata-file)

> - [4. Configure Config File](#4-configure-config-file)

Reviewer: Just noting that the rest of the headers are in sentence case except for these new changes, so let's change them back to sentence case.

2. [Install the packages and dependencies](#c-additional-dependencies) that are required to run the workflow.
3. Ensure that the input single-cell gene expression data are stored as `SingleCellExperiment` objects in RDS files (see more on this in the ["Input data format" section](#2-input-data-format)).
The workflow can directly take as input the `filtered` RDS files downloaded from the [Single-cell Pediatric Cancer Atlas portal](https://scpca.alexslemonade.org/) or the output from the [scpca-nf workflow](https://github.com/AlexsLemonade/scpca-nf), a workflow that can be used to quantify your own single-cell/single-nuclei gene expression data. If working with data from the ScPCA portal, see more information on preparing that data to run the core workflow [here](./additional-docs/working-with-scpca-portal-data.md).
4. [Create a metadata tab-separated value (TSV) file](#3-metadata-file-format) that defines the sample id, library id, and filepath associated with the pre-processed `SingleCellExperiment` files to be used as input for the workflow.

> 5. Configure the config file to adjust the `results_dir` and `project_metadata` parameters to point to your desired results directory and project metadata file that you created in step 4.

Reviewer: ```suggestion
5. Configure the config file to adjust the `results_dir` and `project_metadata` parameters to point to the full path to your desired results directory and project metadata file that you created in step 4.
```
Let's just make sure we specify the full path throughout.

|[View Example Metadata File](https://github.com/AlexsLemonade/scpca-downstream-analyses/blob/main/example-data/project-metadata/example-library-metadata.tsv)|
|---|

## 4. Configure config file

We have provided an [example snakemake configuration file](config/config.yaml), `config/config.yaml` which defines all parameters needed to run the workflow.
Learn more about snakemake configuration files [here](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html).

> You can modify the relevant parameters by manually updating the `config/config.yaml` file using a text editor of your choice.

Reviewer 2: ```suggestion

```

> There are a set of parameters included in the `config/config.yaml` file that will **need to be specified** when running the workflow with your own data.

Reviewer: > I think we should continue to call them Project-specific and Processing parameters as it is more descriptive of what they are.

> Unless I am mistaken, Project specific parameters need to changed to match user's data while the processing parameters do not need to be.

I think this is fine to keep them this way, I would just want to be clear that the project specific parameters are the ones that are required to be changed while the processing ones are the ones that don't need to be changed. To me processing parameters would also mean they might be required to be changed, so I think as long as there is emphasis that those are optional then that's fine.

> These parameters are specific to the project or dataset being processed.

Reviewer 2: ```suggestion

```

> These project-specific parameters can be found under the [`Project-specific parameters` section](./config/config.yaml#L3) of the config file, while the remaining parameters that can be optionally modified are found under the [`Processing parameters` section](./config/config.yaml#L11).

Reviewer: ```suggestion
These required parameters can be found under the [`Required parameters` section](./config/config.yaml#L3) of the config file, while the remaining parameters that can be optionally modified are found under the [`Additional parameters` section](./config/config.yaml#L11).
```
Now that we have broken out the additional parameters, should we rename this to be required vs. additional? To me that is more straight forward and might be less confusing to someone.

Reviewer 2: I think we should continue to call them Project-specific and Processing parameters as it is more descriptive of what they are. 

Unless I am mistaken, Project specific parameters need to changed to match user's data while the processing parameters do not need to be. 

I have made suggestions as to how we can streamline this paragraph to do that.

> These project-specific parameters can be found under the [`Project-specific parameters` section](./config/config.yaml#L3) of the config file, while the remaining parameters that can be optionally modified are found under the [`Processing parameters` section](./config/config.yaml#L11).

Reviewer 2: ```suggestion
The config file contains two sets of parameters:

- **[Project-specific Parameters](./config/config.yaml#L3)** : This set of parameters are for specifying dataset or project related details. 
These parameters are **required** to run the workflow on your data.

- **[Processing Parameters](./config/config.yaml#L11)**: This set of parameters specify configurations for the TODO:say what it is for here
These parameters are set with default values.
You can change them to explore your data but it is optional.

You can modify the relevant parameters by manually updating the `config/config.yaml` file using a text editor of your choice.
```

Reviewer 2: Does this preserve the meaning of the paragraph? Please add any bits of information that I missed. 

If this looks good, I would propagate this change to the clustering and GOI README too.

To run the workflow on your data, modify the following parameters in the `config/config.yaml` file:

| Parameter        | Description |
|------------------|-------------|
| `results_dir` | full path to the directory where output files from running the core workflow will be stored |
| `project_metadata` | full path to your specific project metadata TSV file |
| `mito_file` | full path to a file containing a list of mitochondrial genes specific to the reference genome or transcriptome version used for alignment. By default, the workflow will use the mitochondrial gene list obtained from Ensembl version 104 of the Human transcriptome which can be found in the [`reference-files` directory](./reference-files). |

By default, these parameters point to the [example data](./example-data). 
The two example `_filtered.rds` files were both processed using the [`scpca-nf` workflow](https://github.com/AlexsLemonade/scpca-nf/blob/main/examples/README.md).
Therefore, if you would like to test this workflow using the example data, you can continue to the next step, running the workflow, without modifying the config file.

> The config file also contains processing parameters like cutoffs for minimum genes detected, minimum unique molecular identifiers (UMI) per cell, etc. 

Reviewer: ```suggestion
The config file also contains additional processing parameters like cutoffs for minimum genes detected, minimum unique molecular identifiers (UMI) per cell, etc. 
```

We have set default values for these parameters. 

> Learn more about the [processing parameters](./additional-docs/additional-parameters.md#core-analysis-parameters) and how to modify them.

Reviewer: ```suggestion
Learn more about the [additional processing parameters](./additional-docs/additional-parameters.md#core-analysis-parameters) and how to modify them.
```

By default, these parameters point to the [example data](./example-data). 
The two example `_filtered.rds` files were both processed using the [`scpca-nf` workflow](https://github.com/AlexsLemonade/scpca-nf/blob/main/examples/README.md).
Therefore, if you would like to test this workflow using the example data, you can continue to the next step, running the workflow, without modifying the config file.

The config file also contains additional processing parameters like cutoffs for minimum genes detected, minimum unique molecular identifiers (UMI) per cell, etc. 
We have set default values for these parameters. 
Learn more about the [additional processing parameters](./additional-docs/additional-parameters.md#core-analysis-parameters) and how to modify them.

See the [processing information documentation](./additional-docs/processing-information.md) for more information on the individual workflow steps and how the parameters are used in each of the steps.

|[View Config File](https://github.com/AlexsLemonade/scpca-downstream-analyses/blob/main/config/config.yaml)|
|---|

## 5. Running the workflow

> After you have successfully modified the required parameters in the config file, you can run the snakemake workflow with just the `--cores` and `--use-conda` flags as in the following example: 

Reviewer: ```suggestion
After you have successfully modified the required project-specific parameters in the config file, you can run the snakemake workflow with just the `--cores` and `--use-conda` flags as in the following example: 
```
Sticking with Deepa's comment, we should keep the project/processing language but specify which ones are required to be modified. So I would make this slight modification for clarity.

|[View Example Metadata File](https://github.com/AlexsLemonade/scpca-downstream-analyses/blob/main/example-data/project-metadata/example-library-metadata.tsv)|
|---|

## 4. Configure config file

> We have provided an example [snakemake configuration file](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html), [`config/config.yaml`](config/config.yaml) which defines all parameters needed to run the workflow.

Reviewer 2: The way the sentence is phrased make me think that the link on snakemake configuration will link me to the example file. That combined with the low affordance on links in code blocks makes it difficult to get to the example config file. 

 ```suggestion
We have provided an [example snakemake configuration file](config/config.yaml), `config/config.yaml` which defines all parameters needed to run the workflow.
Learn more about snakemake configuration files [here](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html)
```

Also note that new changes should be merged through a pull request to the `development` branch.
Changes will be pushed to the `main` branch once changes are ready for a new release (per the [release checklist document](.github/ISSUE_TEMPLATE/release-checklist.md)).

## 5. Running the workflow

> You can run the snakemake workflow with just the `--cores` and `--use-conda` flags as in the following example: 

Reviewer: I would add a sentence here that you can run the workflow after you have successfully modified the required parameters in the config file.
